### PR TITLE
Push reconciler events for ingresses only when their ingress class is Kourier

### PR DESCRIPTION
**Pull request by davidor**
We were reacting to all changes in ingresses. This PR changes that so we only react to events produced by those ingresses that have the Kourier ingress class.

### CODE REVIEW

File: pkg/reconciler/ingress/reconciler.go [hunk 1]
```diff
@@ -22,11 +21,9 @@ func (reconciler *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return err
 	}
 
-	kourierIngresses := knative.FilterByIngressClass(ingressAccessors)
-
 	reconciler.EnvoyXDSServer.SetSnapshotForIngresses(
 		nodeID,
-		kourierIngresses,
+		ingressAccessors,
```
Comment (jmprusi) [inline pkg/reconciler/ingress/reconciler.go]: 👀

Comment (davidor) [reply pkg/reconciler/ingress/reconciler.go]: Changed :+1:

**Comment by davidor** [comment]
Don't merge yet, there's a problem. Events are only generated when the ingress class is the Kourier one, but we still need to filter them in the reconciler, otherwise, List() is going to return all of them.

**Comment by jmprusi** [comment]
that's true... check this:

https://github.com/3scale/kourier/pull/112/files#diff-a1c037253522803c5b9cf0e85a17bf7dR158-R165

File: pkg/reconciler/ingress/ingress.go [hunk 2]
```diff
@@ -51,7 +55,17 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 		impl.EnqueueKey("")
 	}
 
-	ingressInformer.Informer().AddEventHandler(controller.HandleAll(enqueueFunc))
+	// Ingresses need to be filtered by ingress class, so Kourier does not
+	// react to nor modify ingresses created by other gateways.
+	ingressInformerHandler := cache.FilteringResourceEventHandler{
+		FilterFunc: func(obj interface{}) bool {
```
Comment (markusthoemmes) [inline pkg/reconciler/ingress/ingress.go]: You can use a helper from Knative to achieve this, see: https://github.com/knative/serving/blob/master/pkg/reconciler/ingress/controller.go#L69

File: pkg/reconciler/ingress/reconciler.go [hunk 3]
```diff
@@ -17,16 +16,14 @@ type Reconciler struct {
 }
 
 func (reconciler *Reconciler) Reconcile(ctx context.Context, key string) error {
-	ingressAccessors, err := reconciler.IngressLister.List(labels.Everything())
+	ingresses, err := reconciler.IngressLister.List(labels.Everything())
 	if err != nil {
 		return err
 	}
 
-	kourierIngresses := knative.FilterByIngressClass(ingressAccessors)
```
Comment (markusthoemmes) [inline pkg/reconciler/ingress/reconciler.go]: I belive we still need to do this. The informer above only scopes which events poke this. The lister here still lists all of the ingresses and we still need to filter that list down.

Comment (davidor) [reply pkg/reconciler/ingress/reconciler.go]: Yes, exactly. I commented this https://github.com/3scale/kourier/pull/117#issuecomment-557127946

I realized after pushing the code. Fixing now.

Comment (davidor) [reply pkg/reconciler/ingress/reconciler.go]: Fixed :+1:

File: pkg/reconciler/ingress/ingress.go (hunk 2 above)
Comment (davidor) [reply pkg/reconciler/ingress/ingress.go]: Nice!

Changed :+1:

**Comment by davidor** [comment]
I fixed what I mentioned in my previous comment: https://github.com/3scale/kourier/pull/117#issuecomment-557127946

**Pull request merged**